compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2584,6 +2584,12 @@ private boolean isXmlAccess(BLangFieldBasedAccess fieldAccessExpr) {
 
     public void visit(BLangIndexBasedAccess indexBasedAccessExpr) {
         // First analyze the variable reference expression.
+        if (indexBasedAccessExpr.expr instanceof  BLangTypedescExpr) {
```

Comment:
Shall we also extract `indexBasedAccessExpr.expr` out to a variable? Seems to be used over and over again.